Mathlib/LinearAlgebra/Matrix/DFT.lean

```diff
@@ -0,0 +1,274 @@
+/-
+Copyright (c) 2020 Mohanad Ahmed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed
+-/
+
+import Mathlib.Algebra.GeomSum
+import Mathlib.Data.Complex.Basic
+import Mathlib.Data.Complex.Exponential
+import Mathlib.Analysis.SpecialFunctions.Complex.Log
+import Mathlib.Analysis.SpecialFunctions.Trigonometric.Basic
+import Mathlib.LinearAlgebra.Matrix.NonsingularInverse
+import Mathlib.LinearAlgebra.Matrix.Circulant
+import Mathlib.LinearAlgebra.Vandermonde
+
+/-!
+# Discrete Fourier Transform (DFT) Matrix and DFT of a (finite) sequence
+
+This file defines the `dft` opertaion on a sequence (also a vecotr) and the DFT operation matrix
+
+## Main definitions
+
+ - `dft v`: given a sequence (v : (Fin n) → ℂ) we can transform it into a sequence (V : (Fin n) →ℂ)
+ such that
+ $$ V [p] = ∑_{k = 0}^{N - 1} e^{-j*2πkp/n} v [k] $$
+ - `idft V` : given a sequence (V : (Fin n) → ℂ) we can transform it into a sequence
+ (v : (Fin n) → ℂ)
+such that
+$$ v [k] = ¼{1}{N}∑_{p = 0}^{n - 1} e^{j*2πkp/N} v [p] $$
+- `Wₙ` : the dft matrix with the `k, p` entry equal to
```

Comment:
I don't think this is like mathlib naming conventions at all

Reply:
Yes but it is the standard notation for the DFT matrix see (https://en.wikipedia.org/wiki/DFT_matrix)

Reply:
sure, but standard for mathlib is to name it something like `dft_matrix` or the like (I'm not sure with the lean4 conventions) - we can note the normal maths names elsewhere